src/Bridge/Symfony/Bundle/Resources/config/http_cache_tags.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="api_platform.http_cache.resource_manager" class="ApiPlatform\Core\HttpCache\ResourceManager" public="false" />
+        <service id="api_platform.http_cache.purger.varnish_client" class="GuzzleHttp\Client" abstract="true" public="false" />
+        <service id="api_platform.http_cache.purger.varnish" class="ApiPlatform\Core\HttpCache\VarnishPurger" public="false" />
+
+        <service id="api_platform.http_cache.listener.response.add_tags" class="ApiPlatform\Core\HttpCache\EventListener\AddTagsListener">
+            <argument type="service" id="api_platform.http_cache.resource_manager" />
+            <argument type="service" id="api_platform.iri_converter" />
+
+            <tag name="kernel.event_listener" event="kernel.response" method="onKernelResponse" />
+        </service>
+
+        <!-- The onKernelView method must be executed after the validation (32) but before the write (64) -->
+        <service id="api_platform.http_cache.listener.response.purge" class="ApiPlatform\Core\HttpCache\EventListener\PurgeListener">
+            <argument type="service" id="api_platform.http_cache.resource_manager" />
+            <argument type="service" id="api_platform.http_cache.purger" />
+
+            <tag name="kernel.event_listener" event="kernel.terminate" method="onKernelTerminate" />
```

Comment:
I suggest to move the purge as close as possible to the transaction (doctrine postFlush IMO)

I understand your point of view to return a response to the user as soon as possible, but this introduce cross concurrency bugs.

Example:
- list all blogs (cache mis)
- post a blog
- list all blog (cache hit when called before the kernel terminate even)

I already have this issue in the my CI (agreed, this is a particular case with a lot of stress tests). but the probability of cross concurrency is higher when:
- you have several varnishes to purge
- you have other kernel.terminate events with an higher priority (sending an email for instance)